tools/pylib/boututils/check_scaling.py

```diff
@@ -0,0 +1,70 @@
+"""Functions for checking the error scaling of MMS or MES results
+
+"""
+
+from numpy import array, isclose, log, polyfit
+
+
+def get_order(grid_spacing, errors):
+    """Get the convergence order of errors over the full range of
+    grid_spacing, and at small spacings
+
+    Parameters
+    ----------
+    grid_spacing : list of float
+        The grid spacing or inverse of number of grid points
+    errors : list of float
+        The error at each grid spacing
+
+    Returns
+    -------
+    tuple of float
+        The first value is the error scaling over the full range of
+        grid spacings; the second value is the scaling over the last
+        two points
+
+    """
+    full_range = polyfit(log(grid_spacing), log(errors), 1)
+
+    small_spacing = log(errors[-2] / errors[-1]) / log(grid_spacing[-2] / grid_spacing[-1])
+
+    return (full_range[0], small_spacing)
+
+
+def check_order(actual_order, expected_order, tolerance=2.e-1):
```

Comment:
I like the idea of a unified api, but I would prefer if the API would take the L2/Linf errors, the grid spacing (optional) and the (absolute) tolerance (also optional) and return success.

NB: The grid spacing is not really needed in (basically) all cases I have seen, as we are going from coarse to fine grid, doubling the number of grid points, thus this is repetetive and can be skipped, as the absolute values don't matter.

Reply:
Splitting it into calculating the error and checking it allows us to do things in-between, such as print the order. Combining the two functions into a one-liner is also trivial, so I much prefer this split